util/EFSOI_Utilities/src/gridio_efsoi.f90

```diff
@@ -0,0 +1,898 @@
+     call nemsio_getfilehead(gfile,iret=iret,vcoord=nems_vcoord)
+  
+     if ( iret /= 0 ) then
+        write(6,*)' gridio:  ***ERROR*** problem reading header ', &
+                  'vcoord, Status = ',iret
+        call stop2(99)
+     endif    
+  
+     if ( idvc == 0 ) then ! sigma coordinate, old file format.
+        ak = zero
+        bk = nems_vcoord(1:nlevs+1,1,1)
+     else if ( idvc == 1 ) then ! sigma coordinate
+        ak = zero
+        bk = nems_vcoord(1:nlevs+1,2,1)
+		
+     else if ( idvc == 2 .or. idvc == 3 ) then ! hybrid coordinate
+   
+        ! AFE      ak = nems_vcoord(1:nlevs+1,1,1)
+        ! AFE      ak = nems_vcoord(1:nlevs+1,2,1)
+        ak = 0.01_r_kind*nems_vcoord(1:nlevs+1,1,1) ! convert to mb
+        bk = nems_vcoord(1:nlevs+1,2,1)
+     else
+        write(6,*)'gridio:  ***ERROR*** INVALID value for idvc=',idvc
+        call stop2(85)
+     end if
+  
+     !==> pressure at interfaces. 
+     do k=1,nlevs+1 
+        pressi(:,k)=ak(k)+bk(k)*tmpgrd 
+     end do 
+     deallocate(ak,bk) 
+  
+  ! === Option TWO, NetCDF === 
+  else if (use_gfs_ncio) then
+	  
+	 call mpi_barrier(mpi_comm_world,ierr)
+
+     call read_vardata(dset, 'pressfc', values_2d,errcode=iret)
+	 
+     if (iret /= 0) then
+        print *,'error reading ps'
+        call stop2(31)
+     endif
+	 
+     psg = 0.01_r_kind*reshape(values_2d,(/nlons*nlats/))
+	 
+     ! Extract surface pressure
+     ! on reduced gaussian grid
+     call regtoreduced(psg,tmpgrd)	 
+	 
+     call read_attribute(dset, 'ak', ak)
+     call read_attribute(dset, 'bk', bk)
+  
+     ! pressure at interfaces
+     do k=1,nlevs+1
+        pressi(:,k) = 0.01_r_kind*ak(nlevs-k+2)+bk(nlevs-k+2)*tmpgrd
+		
+        if (nproc == 0) print *,'netcdf, min/max pressi',k,minval(pressi(:,k)),maxval(pressi(:,k))
+     enddo
+	 
+     deallocate(ak,bk,values_2d)  
+	 
+ ! === Option THREE, other === 	 
+ else
+	 
+    vrtspec = sigdata%ps
+    call sptez_s(vrtspec,psg,1)
+    !==> input psg is ln(ps) in centibars - convert to ps in millibars.
+    psg = 10._r_kind*exp(psg)
+    allocate(ak(nlevs+1),bk(nlevs+1))
+    if (sighead%idvc .eq. 0) then ! sigma coordinate, old file format.
+        ak = zero
+        bk = sighead%si(1:nlevs+1)
+    else if (sighead%idvc == 1) then ! sigma coordinate
+        ak = zero
+        bk = sighead%vcoord(1:nlevs+1,2)
+    else if (sighead%idvc == 2 .or. sighead%idvc == 3) then ! hybrid coordinate
+        bk = sighead%vcoord(1:nlevs+1,2)
+        ak = 0.01_r_kind*sighead%vcoord(1:nlevs+1,1)  ! convert to mb
+    else
+        print *,'unknown vertical coordinate type',sighead%idvc
+        call stop2(32)
+    end if
+
+    do k=1,nlevs+1
+       pressi(:,k)=ak(k)+bk(k)*psg
+       print *,'sigio, min/max pressi',k,minval(pressi(:,k)),maxval(pressi(:,k))
+    enddo
+	
+    deallocate(ak,bk)
+	
+ endif
+ 
+!!! LL
+ if (use_gfs_nemsio) call nemsio_close(gfile,iret=iret)
+ if (use_gfs_ncio)   call close_dataset(dset)   
+  
+  
+  
```

Comment:
@AndrewEichmann-NOAA Can you easily remove these excess empty spaces?

Reply:
done